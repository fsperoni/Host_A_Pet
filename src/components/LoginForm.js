import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import Alert from "./Alert";

/** Login form. */
const LoginForm = ({ login }) => {
  const history = useNavigate();
  const [formData, setFormData] = useState({
    username: "",
    password: "",
  });
  const [formErrors, setFormErrors] = useState([]);

  const handleSubmit = async (evt) => {
    evt.preventDefault();
    const result = await login(formData);
    if (result.success) {
      history("/dashboard");
    } else {
      setFormErrors(result.errors);
    }
  }

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData(form => ({ ...form, [name]: value }));
  }

  return (
    <div className="LoginForm">
      <div className="container col-md-6 offset-md-3 col-lg-4 offset-lg-4">
        <h3 className="mb-3">Log In</h3>
        <div className="card">
          <div className="card-body">
            <form onSubmit={handleSubmit}>
              <div className="form-group">
                <label>Username</label>
                <input className="form-control" autoComplete="username" required
                  value={formData.username} onChange={handleChange} name="username"/>
              </div>
              <div className="form-group">
                <label>Password</label>
                <input className="form-control" type="password" name="password" required
                  value={formData.password} onChange={handleChange} autoComplete="current-password"/>
              </div>

              {formErrors.length ? <Alert type="danger" messages={formErrors} /> : null}

              <button className="btn btn-primary float-right" onSubmit={handleSubmit}>
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  );
}

export default LoginForm;
