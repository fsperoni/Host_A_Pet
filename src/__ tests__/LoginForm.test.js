import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import LoginForm from "../components/LoginForm";

// smoke test
it('renders without crashing', function () {
  render(
    <BrowserRouter>
      <LoginForm login={function () { }} />
    </BrowserRouter>);
});

//snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(
    <BrowserRouter>
      <LoginForm login={function () { }} />
    </BrowserRouter>);
  expect(asFragment()).toMatchSnapshot();
});

it("should have 'Username' and 'Password' in the component.", function () {
  const { getByText } = render(
    <BrowserRouter>
      <LoginForm login={function () { }} />
    </BrowserRouter>);
  expect(getByText("Username")).toBeInTheDocument();
  expect(getByText("Password")).toBeInTheDocument();
});