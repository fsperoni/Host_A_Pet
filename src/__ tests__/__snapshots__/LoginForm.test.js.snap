// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="LoginForm"
  >
    <div
      class="container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
    >
      <h3
        class="mb-3"
      >
        Log In
      </h3>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <form>
            <div
              class="form-group"
            >
              <label>
                Username
              </label>
              <input
                autocomplete="username"
                class="form-control"
                name="username"
                required=""
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label>
                Password
              </label>
              <input
                autocomplete="current-password"
                class="form-control"
                name="password"
                required=""
                type="password"
                value=""
              />
            </div>
            <button
              class="btn btn-primary mt-2 float-right"
            >
              Submit
            </button>
            <p
              class="mt-2"
            >
              Don't have an account? 
              <a
                class=""
                href="/signup"
              >
                Sign Up
              </a>
               today!
            </p>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
